src/OnboardingSPA/components/Sidebar/components/Customize/DesignColorsPanel/index.js

```diff
@@ -114,15 +120,25 @@ const DesignColorsPanel = forwardRef(
 		useEffect( () => {
 			if ( ! customColors ) {
 				const storedCustomColors =
-					currentData.sitegen.homepages.active.color
-						.customColors;
+					currentData.sitegen.homepages.active.color.customColors;
 				if ( storedCustomColors ) {
 					setCustomColors( storedCustomColors );
 				} else {
 					const defaultCustomColors = palettes[ 0 ];
 					setCustomColors( defaultCustomColors );
 				}
 			}
+			if ( ! selectedPalette ) {
+				const storedSelectedPalette =
+					currentData.sitegen.homepages.active.color.selectedPalette;
+				if ( storedSelectedPalette ) {
+					setSelectedPalette( storedSelectedPalette );
+					if ( storedSelectedPalette === 'custom' ) {
+						setShowCustomColors( true );
+						setSelectedCustomColors( true );
+					}
+				}
+			}
```

Comment:
Why do we have so many states like `storedCustomColors` and `storedSelectedPalette custom`? Can we use a single state with a common data structure for storing `default/pre-defined/custom` colors and then just mark that as `default/pre-defined/custom` in another state variable? This way, we will only have to maintain two states.